Sources/templates.swift

```diff
@@ -19,54 +19,48 @@ let templatesListCommand = command(
   },
   outputOption
 ) { onlySubcommand, output in
-  let customTemplates = (try? appSupportTemplatesPath.children()) ?? []
-  let bundledTemplates = (try? bundledTemplatesPath.children()) ?? []
   var outputLines = [String]()
 
-  let printTemplates = { (prefix: String, list: [Path]) in
-    for file in list where file.lastComponent.hasPrefix("\(prefix)-") && file.extension == "stencil" {
+  let printTemplates = { (subcommand: String, path: Path) in
+    guard let files = try? (path + subcommand).children() else { return }
+
+    for file in files where file.extension == "stencil" {
       let basename = file.lastComponentWithoutExtension
-      let idx = basename.index(basename.startIndex, offsetBy: prefix.characters.count+1)
-      let name = basename[idx..<basename.endIndex]
-      outputLines.append("   - \(name)")
+      outputLines.append("   - \(basename)")
     }
   }
 
   let subcommandsToList = onlySubcommand.isEmpty ? allSubcommands : [onlySubcommand]
-  for prefix in subcommandsToList {
-    outputLines.append("\(prefix):")
+  for subcommand in subcommandsToList {
+    outputLines.append("\(subcommand):")
     outputLines.append("  custom:")
-    printTemplates(prefix, customTemplates)
+    printTemplates(subcommand, appSupportTemplatesPath)
     outputLines.append("  bundled:")
-    printTemplates(prefix, bundledTemplates)
+    printTemplates(subcommand, bundledTemplatesPath)
   }
 
   outputLines.append("---")
   outputLines.append("You can add custom templates in \(appSupportTemplatesPath).")
-  outputLines.append("Simply name them 'subcmd-customname.stencil' where subcmd is one of the swiftgen subcommand,")
-  outputLines.append("namely " + allSubcommands.map({"\($0)-xxx.stencil"}).joined(separator: ", ") + ".")
+  outputLines.append("Simply place them in a subfolder of this path with as name one of the swiftgen subcommands,")
+  outputLines.append("namely " + allSubcommands.map({"'\($0)'"}).joined(separator: ", ") + ". Your template must be")
+  outputLines.append("in the right subfolder, and must have as extension '.stencil'. For example, if you created a")
+  outputLines.append("custom template for the strings command, place it in the following path:")
+  outputLines.append("strings/customname.stencil")
   outputLines.append("")
 
   output.write(content: outputLines.joined(separator: "\n"))
 }
 
 private func templatePathCommandGenerator(execute: @escaping (Path, OutputDestination) throws -> Void) -> CommandType {
```

Comment:
Maybe that would a a good occasion to add one or 2 lines of comments here, because it took me a little time to remember what that method was used for before remembering how it's used :wink: